back-end/src/repos/repos.service.ts

```diff
@@ -0,0 +1,87 @@
+import { HttpService, Injectable } from '@nestjs/common';
+import { assign, map, pick, flatMap } from 'lodash';
+import repoConfig from '../../config/repo-data-config';
+import { ReposRepository } from './repos.repository';
+
+@Injectable()
+export class ReposService {
+  constructor(private readonly httpService: HttpService, private readonly repoDB: ReposRepository) {}
+
+  public async getRepo() {
+    const publicRepositories = await this.makeRequestToGitHubLink(repoConfig.repositories);
+    return publicRepositories;
+  }
+
+  private async makeRequestToGitHubLink(repositories) {
+      interface Repos {
```

Comment:
```suggestion
       interface Repos {
        link: string;
        repoType: string;
        repoName: string;
        branch: string;
        token: string;
      }
```

Вот это я бы вынес в отдельную папку с интерфейсами для этого контроллера